tests/common.go

```diff
@@ -3517,75 +3532,117 @@ func DeleteSchedule(backupScheduleName string, clusterName string, orgID string,
+	if len(kubeconfigList) < 2 {
+		return fmt.Errorf("minimum 2 kubeconfigs are required for source and destination cluster")
 	}
 	err := dumpKubeConfigs(configMapName, kubeconfigList)
 	if err != nil {
 		return err
 	}
-	// Register source cluster with backup driver
 	log.InfoD("Create cluster [%s] in org [%s]", SourceClusterName, orgID)
 	srcClusterConfigPath, err := GetSourceClusterConfigPath()
 	if err != nil {
 		return err
 	}
 	log.Infof("Save cluster %s kubeconfig to %s", SourceClusterName, srcClusterConfigPath)
-	sourceClusterStatus := func() (interface{}, bool, error) {
-		err = CreateCluster(SourceClusterName, srcClusterConfigPath, orgID, cloudName, uid, ctx)
-		if err != nil && !strings.Contains(err.Error(), "already exists with status: Online") {
-			return "", true, err
-		}
-		srcClusterStatus, err := Inst().Backup.GetClusterStatus(orgID, SourceClusterName, ctx)
-		if err != nil {
-			return "", true, err
-		}
-		if srcClusterStatus == api.ClusterInfo_StatusInfo_Online {
-			return "", false, nil
-		}
-		return "", true, fmt.Errorf("the %s cluster state is not Online yet", SourceClusterName)
-	}
-	_, err = task.DoRetryWithTimeout(sourceClusterStatus, clusterCreationTimeout, clusterCreationRetryTime)
-	if err != nil {
-		return err
-	}
-	ClusterConfigPathMap[SourceClusterName] = srcClusterConfigPath
-	// Register destination cluster with backup driver
+
 	log.InfoD("Create cluster [%s] in org [%s]", destinationClusterName, orgID)
 	dstClusterConfigPath, err := GetDestinationClusterConfigPath()
 	if err != nil {
 		return err
 	}
 	log.Infof("Save cluster %s kubeconfig to %s", destinationClusterName, dstClusterConfigPath)
-	destClusterStatus := func() (interface{}, bool, error) {
-		err = CreateCluster(destinationClusterName, dstClusterConfigPath, orgID, cloudName, uid, ctx)
-		if err != nil && !strings.Contains(err.Error(), "already exists with status: Online") {
-			return "", true, err
+
+	ClusterConfigPathMap[SourceClusterName] = srcClusterConfigPath
+	ClusterConfigPathMap[destinationClusterName] = dstClusterConfigPath
+
+	clusterCreation := func(clusterCredName string, clusterCredUid string, clusterName string) error {
+		clusterStatus := func() (interface{}, bool, error) {
+			err = CreateCluster(clusterName, ClusterConfigPathMap[clusterName], orgID, clusterCredName, clusterCredUid, ctx)
+			if err != nil && !strings.Contains(err.Error(), "already exists with status: Online") {
+				return "", true, err
+			}
+			srcClusterStatus, err := Inst().Backup.GetClusterStatus(orgID, SourceClusterName, ctx)
+			if err != nil {
+				return "", true, err
+			}
+			if srcClusterStatus == api.ClusterInfo_StatusInfo_Online {
+				return "", false, nil
+			}
+			return "", true, fmt.Errorf("the %s cluster state is not Online yet", SourceClusterName)
 		}
-		destClusterStatus, err := Inst().Backup.GetClusterStatus(orgID, destinationClusterName, ctx)
+		clusterStatus()
```

Comment:
What does it do?

Reply:
This is how the anonymous function gets called.